evm/src/cross_table_lookup.rs

```diff
@@ -623,11 +623,11 @@ fn partial_products<F: Field>(
                 .iter()
                 .map(|c| c.eval_table(trace, i))
                 .collect::<Vec<_>>();
-            partial_prod *= challenge.combine(evals.iter());
+            partial_sum += challenge.combine(evals.iter()).inverse();
```

Comment:
this looks heavy. Every `inverse` with Goldilocks is about 80 multiplications IIRC, better to do something with batch_inverse, I think that's what's done in `lookup.rs`?

Reply:
also I think some methods may be partial duplicates now of the logUp implementation used in lookups for range-checks. May be nice to unify things if possible under common utility methods

Reply:
I added batch inversion.

Mutualizing methods is actually non-trivial, since the logic between CTLs and range-checks is not identical. This should be done at some point though (this would make CTL batching much easier, as well as filters for range-checks), but it's complex enough to be a PR on its own.